src/frontends/lean/elaborator.cpp

```diff
@@ -2420,18 +2421,12 @@ expr elaborator::visit_field(expr const & e, optional<expr> const & expected_typ
         try {
             full_fname = field_to_decl(e, s, new_s_type);
         } catch (elaborator_exception & ex2) {
-            throw nested_elaborator_exception(ex2.get_main_expr(), ex1, ex2.pp());
+            throw nested_elaborator_exception(ex2.get_pos(), ex1, ex2.pp());
         }
     }
     expr proj  = copy_tag(e, mk_constant(full_fname));
     expr new_e = copy_tag(e, mk_app(proj, copy_tag(e, mk_as_is(s))));
-    expr r = visit(new_e, expected_type);
-    if (is_app(r)) {
-        /* r may be a sorry macro due to error recovery*/
-        if (auto pos = get_field_notation_field_pos(e))
-            save_identifier_info(app_fn(r), pos);
-    }
-    return r;
+    return visit(new_e, expected_type);
```

Comment:
Do we still get identifier information for the projection? That is, given `t^.reverse`, where `t : list nat`, do we get `list.reverse` for `^.reverse`?